### DISCUSSION TITLE
Hook for formatter: idea wanted

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#1585] [Q&A]
we have part of our models generated from data in a database.
as langium does not support serialization (with or without formatting)
we currently use this approach

1. write a text file with the models using template strings
2. parse these and call and apply formatter on it.

we wonder if we could spare 2 and call the formatter only if needed e.g.
- in interactive usecase / vscode
now the question would be when to do this
- before sending the files to vscode / materializing the workspace
- somewhere client or server side in vscode

would be thankful for some hints or ideas

**Comment by msujew** [comment]
I haven't thought too much about this, but does `TextDocument.applyEdits` in combination with the formatter work for you? You don't need to write the file to disk, you can just create an in-memory document for the parsed text and run it through the formatter. You can then store the formatted document on disk.

**Reply by cdietrich** [reply]
We need to figure out what we have in hand. Current we have a background process that transferrs all files (not dsl only) to disk. So nothing that is inited by language server or talks to it
